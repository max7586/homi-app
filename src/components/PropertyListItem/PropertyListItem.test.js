import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import PropertyListItem from './PropertyListItem';
import { BrowserRouter } from 'react-router-dom'
// import { Link } from 'react-router-dom'

describe('PropertyListItem component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <BrowserRouter>
        <PropertyListItem property={{id:2,title:'test',content:''}}/>
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<BrowserRouter>
        <PropertyListItem property={{id:2,title:'test',content:''}}/>
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });


});