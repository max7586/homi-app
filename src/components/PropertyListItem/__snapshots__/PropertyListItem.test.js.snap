// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyListItem component renders the UI as expected 1`] = `
<a
  className="PropertyListItem"
  href="/property/2"
  onClick={[Function]}
>
  <div
    className="PropertyListItem__image"
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
  />
  <div
    className="PropertyListItem__details"
  >
    <div
      className="PropertyListItem__text"
    >
      <h2
        className="PropertyListItem__heading"
      >
        test
      </h2>
      <p
        className="PropertyListItem__description"
      >
        
      </p>
      <p
        className="PropertyListItem__userId"
      />
    </div>
    <div
      className="PropertyListItem__reviews"
    >
      <span
        className="PropertyStarRating"
      />
      <span
        id="PropertyListItem__review-count"
      >
        based on undefined reviews
      </span>
    </div>
  </div>
</a>
`;
